sdk/messaging/azeventhubs/checkpoint_store.go

```diff
@@ -11,19 +11,19 @@ import (
 )
 
 // CheckpointStore is used by multiple consumers to coordinate progress and ownership for partitions.
-type CheckpointStore interface {
+type CheckpointStore struct {
```

Comment:
Thinking out loud, given the below conversation around `NewBlobStore()`, I'm reconsidering if this change is worth it (I didn't know that we return types that satisfy this interface).  With this change, it means that the `BlobStore` type can't expose any APIs other than what `CheckPointStore` exposes, since we can't return an actual `BlobStore` instance; it essentially tightly couples `BlobStore` (and other future types) to the `CheckpointStore` type which seems unfortunate.

CC @JeffreyRichter to weigh in on this as well.

Reply:
It's a valid concern, for sure.

I think we can ignore this, however, because the majority of the type this is just an opaque type, so we wouldn't offer additional functionality on the `CheckpointStore` implementors. 

I could see us thinking about extending the overall CheckpointStore definition at some point, however, so having it be a struct with func()'s instead of an interface should cover what I hope is the actual common extension point.

Reply:
Reverted!